### DISCUSSION TITLE
How training data looks like for Relation extraction component?

### DISCUSSION THREAD

**Discussion by Ibrokhimsadikov** [explosion/spaCy#8432] [Help: Coding & Implementations] [answered]
Hello dear SpaCy funs,

Quick question regarding very important tutorial on Relation extraction component. Everything seems to be pretty clear in terms development and architecture of the trainable component, but it is still blur for me how training data looks like for relation  extraction. For ner it is pretty clear and very well documented, but what about Relation extraction. Please, help me to understand and discover it. in rel_component tutorial they give us overall prodigy annotated json, but it  has parsing and tokens as well, seemed  they re-trained everything on the one go, what if I am only interested in ner and relation extraction part.

Thank you for your time

**Comment by svlandeg** [comment, marked as answer]
Hi!

The REL component is not built-in functionality provided by spaCy, but we wanted to offer the example project as a sort of starting point to implement this. For real-world use-cases, it would be good to experiment with additional features etc on top of the current implementation.

The [video](https://www.youtube.com/watch?v=8HL-Ap5_Axo) describes the annotation format around minute 26:

![afbeelding](https://user-images.githubusercontent.com/8796347/122527679-d8819f00-d01b-11eb-993a-d9645a269b13.png)

The custom component uses a custom attribute `doc._.rel` that consists of a dictionary, linking entity start offsets to another dictionary of mapping REL labels to their gold value `0.0` or `1.0`.

This is also described here: https://spacy.io/usage/layers-architectures#component-rel-attribute

**Reply by Ibrokhimsadikov** [reply]
Thank you dear @svlandeg  for your quick response and it is honor for me to get such detailed response from you. Thank you explosion team for their hard work.
